### PR TITLE
refactor(experimental): create @solana/rpc-transformers package

### PR DESCRIPTION
**Pull request by lorisleiva**
The PR extract the parameter and response transformers that can be used for both `RpcApis` and `RpcSubscriptionsApis` within a new `@solana/rpc-transformers` package.

Notice that, since the both `RpcApis` and `RpcSubscriptionsApis` will end up depending on this package, the response transformers was decoupled from them. Instead of hard-coding the APIs it support and generic specific transformers for both of them, this package now exports a generic function that accepts any `TApi` definition.

In the next couple of PRs, new `rpc-api` and `rpc-subscriptions-api` packages will import these generic functions and fill them with specific parameters.

RPC packages reshuffle@2x (6)

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2152?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#2153** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2153?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2152** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2152?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2151** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2151?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2150** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2150?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2149** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2149?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2148** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2148?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2147** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2147?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2146** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2145** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2144** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2152?utm_source=stack-comment).

**Review by steveluscher — APPROVED** [review]
Added via Giphy

File: packages/rpc-transformers/src/params-transformer-options-object-position-config.ts
Comment (buffalojoec) [inline packages/rpc-transformers/src/params-transformer-options-object-position-config.ts:1]: The PR diagram seems to imply that this is a generic transformer library, but this file is most definitely Solana-specific. Should this be Solana-only?

Comment (lorisleiva) [reply packages/rpc-transformers/src/params-transformer-options-object-position-config.ts:1]: That's a good point. I should probably split that file in two and add push them up one level. I'll make a note to do that in a follow up PR.

**Comment by lorisleiva** [comment]
✅ This PR was merged as part of https://github.com/solana-labs/solana-web3.js/pull/2168.

**Pull request closed without merging**